labs_api/insights/insights.py

```diff
@@ -9,44 +11,22 @@
     OmenMarket,
     OmenSubgraphHandler,
 )
+from prediction_market_agent_tooling.tools.caches.db_cache import db_cache
 from prediction_market_agent_tooling.tools.langfuse_ import (
     get_langfuse_langchain_config,
     observe,
 )
-from prediction_market_agent_tooling.tools.tavily_storage.tavily_models import (
-    TavilyResponse,
-    TavilyStorage,
-)
-from prediction_market_agent_tooling.tools.tavily_storage.tavily_storage import (
-    TavilyStorage,
-    tavily_search,
-)
+from prediction_market_agent_tooling.tools.tavily.tavily_models import TavilyResponse
+from prediction_market_agent_tooling.tools.tavily.tavily_search import tavily_search
 from prediction_market_agent_tooling.tools.utils import (
     LLM_SUPER_LOW_TEMPERATURE,
     utcnow,
 )
 
-from labs_api.insights.insights_cache import (
-    MarketInsightsResponse,
-    MarketInsightsResponseCache,
-)
-
-
-# Don't observe the cached version, as it will always return the same thing that's already been observed.
-def market_insights_cached(
-    market_id: HexAddress, cache: MarketInsightsResponseCache
-) -> MarketInsightsResponse:
-    """Returns `market_insights`, but cached daily."""
-    if (cached := cache.find(market_id)) is not None:
-        return cached
-
-    else:
-        new = market_insights(market_id)
-        if new.has_insights:
-            cache.save(new)
-        return new
+from labs_api.insights.insights_models import MarketInsightsResponse
 
 
+@db_cache(max_age=timedelta(days=3))
```

Comment:
I thought that in APIs, it makes sense to observe only the first call -- no benefit in observing every call from potentially many many users on frontend. That's why the db_cache decorator is above observe decorator.